auto_nag/scripts/spambug.py

```diff
@@ -0,0 +1,76 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from auto_nag.bugbug_utils import get_bug_ids_classification
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.utils import nice_round
+
+
+class SpamBug(BzCleaner):
+    def __init__(self):
+        super().__init__()
+
+    def description(self):
+        return "[Using ML] Detect spam bugs"
+
+    def columns(self):
+        return ["id", "summary", "confidence", "autofixed"]
+
+    def sort_columns(self):
+        return lambda p: (-p[2], -int(p[0]))
+
+    def get_bz_params(self, date):
+        start_date, end_date = self.get_dates(date)
+
+        return {
+            "include_fields": ["id", "groups", "summary"],
+            # Ignore closed bugs.
+            "bug_status": "__open__",
+        }
+
+    def get_bugs(self, date="today", bug_ids=[]):
+        # Retrieve the bugs with the fields defined in get_bz_params
+        raw_bugs = super().get_bugs(date=date, bug_ids=bug_ids, chunk_size=7000)
+
+        if len(raw_bugs) == 0:
+            return {}
+
+        # Extract the bug ids
+        bug_ids = list(raw_bugs.keys())[:1]
+
+        # Classify those bugs
+        bugs = get_bug_ids_classification("SpamBug", bug_ids)
+
+        results = {}
+
+        for bug_id in sorted(bugs.keys()):
+            bug_data = bugs[bug_id]
+
+            if not bug_data.get("available", True):
+                # The bug was not available, it was either removed or is a
+                # security bug
+                continue
+
+            if not {"prob", "index"}.issubset(bug_data.keys()):
+                raise Exception(f"Invalid bug response {bug_id}: {bug_data!r}")
+
+            bug = raw_bugs[bug_id]
+            prob = bug_data["prob"]
+            index = bug_data["index"]
+
+            results[bug_id] = {
```

Comment:
Only add results which have a confidence to be spam higher than the confidence threshold set in the config (`self.get_config("confidence_threshold")`)